### PR TITLE
Pulled in only compression related changes from Andrei's commit 0b5b4…

### PR DESCRIPTION
**Pull request by aoblebea**
…79fe5a74db074a1b3e6dcbdca8e22fa8749

Users can specify "tiledb_compression_type" and "tiledb_compression_level" fields in loader files, which will now pass the values to TileDB.

### CODE REVIEW

File: src/main/cpp/include/loader/tiledb_loader_text_file.h [hunk 1]
```diff
@@ -175,7 +175,7 @@ class LineBasedTextFile2TileDBBinary : public File2TileDBBinaryBase {
                             treat_deletions_as_intervals,
                             parallel_partitions, noupdates, close_file) {
     vid_mapper.build_tiledb_array_schema(m_array_schema, "dummy",
-                                         false, false);
+                                         false, 0, 0, false);
```
Comment (nalinigans) [inline src/main/cpp/include/loader/tiledb_loader_text_file.h:178]: Use `TILEDB_NO_COMPRESSION` or `TILEDB_GZIP` for compression and `0` and `TILEDB_COMPRESSION_LEVEL_GZIP` based on the compression default.

File: src/main/cpp/src/utils/vid_mapper.cc [hunk 2]
```diff
@@ -416,18 +418,25 @@ const {
   types.push_back(std::type_index(typeid(int64_t)));
 
   std::vector<int> compression;
```
Comment (nalinigans) [inline src/main/cpp/src/utils/vid_mapper.cc:420]: Call this field compression_types instead of compression too to be consistent with compression_levels

**Review by nalinigans — COMMENTED** [review]
Can you add tests passing compression types/levels explicitly? It can be from `tests/run.py` and/or `tests/test_tools.sh` and/or `tests/src/test/cpp/src/test_config.cc`.

Also, allow tools/src/vcf2genomicsdb_init.cc to optionally specify compression type/level and add that test to `tests/test_tools.sh`?

**Comment by kgururaj** [comment]
> Can you add tests passing compression types/levels explicitly? It can be from `tests/run.py` and/or `tests/test_tools.sh` and/or `tests/src/test/cpp/src/test_config.cc`.
> 
> Also, allow tools/src/vcf2genomicsdb_init.cc to optionally specify compression type/level and add a test for this functionality to `tests/test_tools.sh`?

Should the tests be in GenomicsDB or TileDB? Isn't GenomicsDB simply passing a parameter to the TileDB API?

**Comment by nalinigans** [comment]
> > Can you add tests passing compression types/levels explicitly? It can be from `tests/run.py` and/or `tests/test_tools.sh` and/or `tests/src/test/cpp/src/test_config.cc`.
> > Also, allow tools/src/vcf2genomicsdb_init.cc to optionally specify compression type/level and add a test for this functionality to `tests/test_tools.sh`?
> 
> Should the tests be in GenomicsDB or TileDB? Isn't GenomicsDB simply passing a parameter to the TileDB API?

Yes, the act of actually testing the (de)compression is in TileDB. But, the functionality to configure compression types/levels in GenomicsDB should be tested in GenomicsDB at some level.

**Review by nalinigans — APPROVED** [review]
Looks good to me. @mlathara, if you are ok, we can merge these changes.

File: tests/test_tools.sh [hunk 3]
```diff
@@ -208,6 +233,17 @@ create_template_loader_json
...
+TILEDB_COMPRESSION_ZLIB=1
+TILEDB_COMPRESSION_ZSTD=2
+TILEDB_COMPRESSION_LZ4=3
+create_compression_template_loader_json $TILEDB_COMPRESSION_ZLIB -1
+run_command_and_check_results "vcf2genomicsdb_init -w $WORKSPACE -S $SAMPLE_DIR -o -t $COMPRESSION_TEMPLATE" 2 85 24 85 "#18"
+create_compression_template_loader_json $TILEDB_COMPRESSION_ZSTD 5 1
+run_command_and_check_results "vcf2genomicsdb_init -w $WORKSPACE -S $SAMPLE_DIR -o -t $COMPRESSION_TEMPLATE" 2 85 24 85 "#19"
+create_compression_template_loader_json $TILEDB_COMPRESSION_LZ4 -1 1
+run_command_and_check_results "vcf2genomicsdb_init -w $WORKSPACE -S $SAMPLE_DIR -o -t $COMPRESSION_TEMPLATE" 2 85 24 85 "#20"
+
```
Comment (mlathara) [inline tests/test_tools.sh:246]: Small request: we don't have any easy way of testing that these compression types/levels are being passed correctly all the way through to tiledb with these tests. Could we add one test that sets an invalid compression type and check that we get an error of some sort?

**Review by mlathara — APPROVED** [review]
lgtm

**Pull request merged**